crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -165,6 +178,19 @@ pub fn derive_bundle(input: TokenStream) -> TokenStream {
                 #(#field_get_components)*
             }
         }
+
+        #[cfg(test)]
```

Comment:
Why is the `#[cfg(test)]` (neccessary) here?
Shouldn't that check always happen, regardless of whether you run tests?

Reply:
Looking at the CI Failure, the added doc-test didn't fail to compile, because of the `#[cfg(test)]`.

Reply:
Ideally, I want to prevent any code-gen when you're actually building a binary.

Reply:
Though it may not be necessary... 🤔

Reply:
Nvm, you're correct, we should remove this otherwise you'll get inconsistent behavior haha.